org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/BindingResolver.java

```diff
@@ -30,13 +30,14 @@
  * </p>
  *
  * @see AST#getBindingResolver
+ * @since 3.38
  */
-class BindingResolver {
+public class BindingResolver {
```

Comment:
I don't think making all this public is welcome here; particularly since this class is really likely to evolve together with the language, it will be modified frequently.

Reply:
We need to extend this class, and I can't see a way to do that without making it accessible. This class acts more as an interface than a class, given the method implementations are similar to what you'd see for a default method implementation on an interface.

Reply:
A fragment, as we have now, allows that. I don't think there is a less intrusive way to provide extensibility here than mandating usage of a fragment.